robot/impl/local_robot.go

```diff
@@ -520,7 +525,7 @@ func (r *localRobot) getDependencies(
 	var needUpdate bool
 	for _, dep := range r.manager.resources.GetAllParentsOf(rName) {
 		if node, ok := r.manager.resources.Node(dep); ok {
-			if r.lastWeakDependentsRound <= node.UpdatedAt() {
+			if r.lastWeakDependentsRound.Load() <= node.UpdatedAt() {
```

Comment:
I realized I suggested making this atomic, but I do want to get a better sense of why we need to do this. When do `getDependencies` and `updateWeakDependents` run in concurrent goroutines?

Reply:
Only during a timeout. Because we run resources sequentially, in the happy case there should only ever be one access at a time. But in cases where a resource times out, the goroutine is still out there so we can't prevent it from potentially running `updateWeakDependents` while another resource is doing the same.